unit-tests/src/test/java/org/eclipse/collections/impl/bimap/mutable/HashBiMapTest.java

```diff
@@ -102,4 +105,22 @@ public void withKeysValues()
         Assert.assertSame(map, map44);
         Assert.assertSame(map3, map4);
     }
+
+    @Test
+    public void forcePut_inverseKeyPreservation()
+    {
+        Key key1 = new Key("1");
+        Key key2 = new Key("2");
+
+        Key value1 = new Key("abc");
+
+        Key duplicateOfKey1 = new Key("1");
+
+        HashBiMap<Key, Key> biMap = this.newMapWithKeysValues(key1, value1, key2, new Key("xyz"));
+
+        biMap.forcePut(duplicateOfKey1, new Key("xyz"));
+
+        Assert.assertSame(key1, Iterate.getFirst(biMap.entrySet()).getKey());
+        Assert.assertSame(key1, Iterate.getFirst(biMap.inverse().entrySet()).getValue());
+    }
```

Comment:
Repeating my comment on the latest commit.

There are no assertions about `"abc"` or `"xyz"`. If it's important to make assertions about values (or the Bimap's invere's keys), then they should be added here. If it's not important to test separately, then it's unnecessary to use BiMap<Key, Key> and you could simplify to BiMap<Key, String>.